engine/src/flutter/shell/platform/android/image_external_texture.h

```diff
@@ -35,10 +35,14 @@ namespace flutter {
 ///
 class ImageExternalTexture : public flutter::Texture {
  public:
+  /// Whether the last image should be reset when the context is destroyed.
+  enum class ImageLifecycle { kReset, kKeepAlive };
+
   explicit ImageExternalTexture(
       int64_t id,
       const fml::jni::ScopedJavaGlobalRef<jobject>& image_texture_entry,
-      const std::shared_ptr<PlatformViewAndroidJNI>& jni_facade);
+      const std::shared_ptr<PlatformViewAndroidJNI>& jni_facade,
+      const ImageLifecycle lifecycle = ImageLifecycle::kReset);
```

Comment:
Here and elsewhere I would probably leave this without a default value. its important that we don't ever forget to pass it, as IIRC there is not a safe default: that is, not disposing the image when the surface is an error and disposing the image when the surface is retained leads to a blank screen.